cmake/Modules/FindQhull.cmake

```diff
@@ -61,34 +109,35 @@ endif()
 
 get_filename_component(QHULL_LIBRARY_DEBUG_NAME "${QHULL_LIBRARY_DEBUG}" NAME)
 
-if(QHULL_INCLUDE_DIR AND QHULL_LIBRARY)
-
-  # Include directories
-  set(QHULL_INCLUDE_DIRS ${QHULL_INCLUDE_DIR})
-  unset(QHULL_INCLUDE_DIR)
-  mark_as_advanced(QHULL_INCLUDE_DIRS)
-
-  # Libraries
-  set(QHULL_LIBRARIES optimized ${QHULL_LIBRARY} debug ${QHULL_LIBRARY_DEBUG})
-  unset(QHULL_LIBRARY)
-  unset(QHULL_LIBRARY_DEBUG)
-  mark_as_advanced(QHULL_LIBRARIES)
-
-endif()
-
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Qhull
   FOUND_VAR QHULL_FOUND
-  REQUIRED_VARS QHULL_LIBRARIES QHULL_INCLUDE_DIRS
+  REQUIRED_VARS QHULL_LIBRARY QHULL_INCLUDE_DIR
 )
 
 if(QHULL_FOUND)
   set(HAVE_QHULL ON)
-  if(NOT QHULL_USE_STATIC)
-    add_definitions("-Dqh_QHpointer")
-    if(MSVC)
-      add_definitions("-Dqh_QHpointer_dllimport")
+  add_library(QHULL::QHULL ${QHULL_LIBRARY_TYPE} IMPORTED)
+  set_target_properties(QHULL::QHULL PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${QHULL_INCLUDE_DIR}")
+  set_property(TARGET QHULL::QHULL APPEND PROPERTY IMPORTED_CONFIGURATIONS "RELEASE")
+  set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "CXX")
+  set_target_properties(QHULL::QHULL PROPERTIES COMPILE_DEFINITIONS "-Dqh_QHpointer")
+  if(MSVC)
+    set_target_properties(QHULL::QHULL PROPERTIES COMPILE_DEFINITIONS "--Dqh_QHpointer_dllimport")
```

Comment:
`--Dqh_QHpointer_dllimport` was the double `-` intentional? Also, does `cl` (windows compiler accept) definitions prefixed with `-D`. I think it's supposed to be `/D`.
https://docs.microsoft.com/en-us/cpp/build/reference/d-preprocessor-definitions?view=vs-2019
Help, windows people @larshg , @Morwenn

Reply:
I believe that `target_compile_definitions(QHULL::QHULL PUBLIC qh_QHpointer)` is the preferred portable way to achieve this without having to care about compiler-specific syntax.

Reply:
It was indeed a mistake, I will add @Morwenn solution !

Reply:
Actually there is one issue with this:
 `target_compile_definitions`  only support `INTERFACE` items on IMPORTED targets, but this isn't available on old cmake version (like the 3.5.1 on Ubuntu 16.04).

So compiler specific flags may still be needed...

Reply:
Oh, that's unfortunate :/

Reply:
Indeed :( , I'm still looking for a better solution.

I don't know how QHULL works but does the definition `qh_QHpointer_dllimport` replace `qh_QHpointer` on MSVC, or both definitions are needed ?

Reply:
For now, I've kept the `if(MSVC)`, and updated the code to use `INTERFACE_COMPILE_DEFINITIONS` I've also removed  `-D` prefix I think that this wasn't necessary.